tests/metadata_generator_test.py

```diff
@@ -0,0 +1,258 @@
+    xml = tostring(root, pretty_print=True)
+    assert xml.decode('utf-8') == ('<root>\n'
+                                   '  <dog role="pet">Harold</dog>\n'
+                                   '</root>\n')
+
+
+def test_create_dict_subelement_no_extra_data():
+    root = Element('root')
+    mg.create_dict_subelement(root, 'dog', 'Harold')
+    xml = tostring(root, pretty_print=True)
+    assert xml.decode('utf-8') == ('<root>\n'
+                                   '  <dog>Harold</dog>\n'
+                                   '</root>\n')
+
+
+def test_create_dict_subelement_degree_order_handling():
+    # Order should be name, level, discipline, grantor.
+    root = Element('root')
+    mg.create_dict_subelement(root,
+                              'degree',
+                              content={'discipline': 'Chemistry',
+                                       'level': 'Masters',
+                                       'grantor': 'UNT',
+                                       'name': 'foo'})
+    xml = tostring(root, pretty_print=True)
+    assert xml.decode('utf-8') == ('<root>\n'
+                                   '  <degree>\n'
+                                   '    <name>foo</name>\n'
+                                   '    <level>Masters</level>\n'
+                                   '    <discipline>Chemistry</discipline>\n'
+                                   '    <grantor>UNT</grantor>\n'
+                                   '  </degree>\n'
+                                   '</root>\n')
+
+
+def test_create_dict_subelement_content_dict_not_degree():
+    root = Element('root')
+    content = {'foo': 'one', 'bar': 'two'}
+    mg.create_dict_subelement(root, 'stuff', content=content)
+    # Check the content keys are children of the stuff element.
+    assert root[0].tag == 'stuff'
+    assert len(list(root[0])) == 2
+    for key, value in content.items():
+        assert root[0].find(key).text == value
+
+
+def test_highwiredict2xmlstring():
+    issue = hs.CitationIssue(content='1')
+    title = hs.CitationTitle(content='Important paper')
+    elements = [issue, title]
+    xml = mg.highwiredict2xmlstring(elements, ordering=['citation_title',
+                                                        'citation_issue'])
+    print(xml.decode('utf-8'))
+    assert xml.decode('utf-8') == ('<?xml version="1.0" encoding="UTF-8"?>\n'
+                                   '<metadata>\n'
+                                   '  <meta content="Important paper"'
+                                   ' name="citation_title"/>\n'
+                                   '  <meta content="1" name="citation_issue"/>\n'
+                                   '</metadata>\n')
+
+
+def test_breakString_shorter_than_width_with_offset():
+    line = mg.breakString('Hello world', width=79, firstLineOffset=0)
+    assert line == 'Hello world'
+
+
+def test_breakString_longer_than_width_with_offset():
```

Comment:
I think this test can be improved. If I understand the unit under test correctly, the `firstLineOffset` argument exists to break the first line at a point _before_ it would normally be broken. But with only one space in the test string, and the `firstLineOffset` being less than the length of the last word, we really don't see what's _different_ about this test. Maybe we can change the string to be something like this: "Hi, it's me!", which would normally be split like this: "Hi, it's\n me!", but with the given args, should come out to this, if I'm correct: "Hi,\n it's me!"

Reply:
The function is meant to limit line lengths written, and we use it specifically for writing ANVL formatted files that look like a key/value sort of thing:
```
title: Important Paper
description: This is a description that is long, surpassing allowed line length,
 and needs to be wrapped onto another line.
```
The `firstLineOffset` is used to adjust the number of chars we allow from our string to go on the first line of a broken string to allow for the "key" to be included with the first broken off part of the string. I am not sure of the relevance of the last word vs. the `firstLineOffset`--more I think the importance is to have a sample data where the offset (4) and the length of the string before the whitespace char ('Hello': 5) is under the width allowed, so the string is split onto two lines. If you are wanting a greater focus on `firstLineOffset`, I could adjust the `width` to `12` to emphasize that even though the string 'Hello world' is less than the `width` argument, the string will still be broken because it is not less than the `width` - `firstLineOffset` actual width. I believe the code is run through the same in that case though. I may be misunderstanding your concern, so explain further in person if you feel like I am missing your point.